dbms/src/Flash/Mpp/MinTSOScheduler.cpp

```diff
@@ -153,9 +155,12 @@ void MinTSOScheduler::scheduleWaitingQueries(MPPTaskManager & task_manager)
         /// schedule tasks one by one
         while (!query_task_set->waiting_tasks.empty())
         {
-            auto task = query_task_set->waiting_tasks.front();
-            if (!scheduleImp(current_query_id, query_task_set, task, true))
+            auto task = query_task_set->task_map.find(query_task_set->waiting_tasks.front())->second;
+            if (task != nullptr && !scheduleImp(current_query_id, query_task_set, task, true))
+            {
+                query_task_set->waiting_tasks.pop(); /// it should be pop from the waiting queue, as the task is scheduled as the exceeded state.
```

Comment:
Seems we can not remove the waiting task for all the cases, consider that a task is not min-tso, and it is not scheduled due to soft limit, then why remove it from `waiting_tasks`?